plugins/jabra-gnp/fu-jabra-gnp-device.c

```diff
@@ -1100,6 +1264,14 @@ fu_jabra_gnp_device_set_quirk_kv(FuDevice *device,
 	return FALSE;
 }
 
+static gboolean
+fu_jabra_gnp_device_ready(FuDevice *device, GError **error)
+{
+	/* apparently setting up the child device fails without this function being explicitly
+	 * defined */
```

Comment:
eigh? This smells like a workaround.

Reply:
So, lets dig in a bit. `FuJabraGnpDevice` derives from `FuUsbDevice` which in turn derives from `FuUdevDevice`.

`FuUsbDevice` hooks `device_class->ready` as `fu_usb_device_ready`. In that we're checking for emulated, getting the interfaces for the device and adding fallback icons. Could the issue be that a "child device" isn't really a `FuUsbDevice` (and thus doesn't have interfaces)?

I think the "proper" way to fix this would be to define a new GObject of "`FuJabraGnpChildDevice`" (e.g. adding `fu-jabra-gnp-child-device.[c|h]`) -- I think that's going to make some of the other code less complicated too.

Reply:
We did that.